### PR TITLE
Apply PHP 8.0 Syntax and constructor promotion

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.0, php 8.0 syntax can constuctor promotion can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

**Comment by samsonasik** [comment]
@Ocramius I think it is ready.

**Comment by Ocramius** [comment]
Yes, but #66 takes precedence, and I don't have a computer to work with, these days

File: src/Check/GuzzleHttpService.php [hunk 1]
```diff
@@ -62,7 +59,7 @@ public function __construct(
         array $headers = [],
         array $options = [],
         $statusCode = 200,
-        $content = null,
+        protected ?string $content = null,
```
Comment (Ocramius) [inline src/Check/GuzzleHttpService.php]: Not sure about this introduced type declaration

File: src/Check/HttpService.php [hunk 2]
```diff
@@ -13,42 +13,27 @@
...
-    /** @var int */
-    protected $content;
-
     /**
      * @param string $host       Host name or IP address to check.
      * @param int    $port       Port to connect to (defaults to 80)
      * @param string $path       The path to retrieve (defaults to /)
      * @param int    $statusCode (optional) Expected status code
-     * @param null   $content    (optional) Expected substring to match against the page content.
+     * @param string $content    (optional) Expected substring to match against the page content.
```
Comment (Ocramius) [inline src/Check/HttpService.php]: `string|null`?

File: src/Check/OpCacheMemory.php [hunk 3]
```diff
@@ -19,10 +20,8 @@ class OpCacheMemory extends AbstractMemoryCheck
 {
     /**
      * OpCache information
-     *
-     * @var array
      */
-    private $opCacheInfo;
+    private array|bool $opCacheInfo = false;
```
Comment (Ocramius) [inline src/Check/OpCacheMemory.php:24]: Perhaps we can change approach here: require this property to be a well-formed `array`, and populating it only when the check succeeded? 

Allowing `false` here seems wrong.

File: src/Check/StreamWrapperExists.php [hunk 4]
```diff
@@ -66,9 +65,7 @@ public function check()
     {
         $missingWrappers   = [];
         $availableWrappers = stream_get_wrappers();
-        array_walk($availableWrappers, function ($v) {
-            return strtolower($v);
-        });
+        array_walk($availableWrappers, static fn($v): string => strtolower($v));
```
Comment (Ocramius) [inline src/Check/StreamWrapperExists.php]: Wait... is this even doing anything? D:

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Check/StreamWrapperExists.php (hunk 4 above)
Comment (samsonasik) [reply src/Check/StreamWrapperExists.php]: It doesn't do anything, as the value is not referenced, I removed it https://github.com/laminas/laminas-diagnostics/pull/68/commits/c900a1cd13288e9e94aa2e5c5fb35da04c2ce662

File: src/Check/OpCacheMemory.php (hunk 3 above)
Comment (samsonasik) [reply src/Check/OpCacheMemory.php:24]: that won't work, as it assigned to `opcache_get_status()` in `check()` method, which can be `false`, mark only `array` will cause Fatal error, see https://3v4l.org/I6hHI

File: src/Check/HttpService.php (hunk 2 above)
Comment (samsonasik) [reply src/Check/HttpService.php]: Updated to use `string|null` https://github.com/laminas/laminas-diagnostics/pull/68/commits/b075a902d53376d034b6b056eb431c1426754f07

File: src/Check/GuzzleHttpService.php (hunk 1 above)
Comment (samsonasik) [reply src/Check/GuzzleHttpService.php]: Ok, I removed type declaration https://github.com/laminas/laminas-diagnostics/pull/68/commits/08e2127222a2e1294f3cdb61f880509132f969ad

**Comment by samsonasik** [comment]
@Ocramius All green 🎉

File: src/Check/OpCacheMemory.php (hunk 3 above)
Comment (Ocramius) [reply src/Check/OpCacheMemory.php:24]: OK, let's roll with the current implementation, for now :)

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik!

**Pull request merged**